File: ka_start/ka-start.py
import boto3
from botocore.vendored import requests
import json
client = boto3.client('kinesisanalytics')

def lambda_handler(event, context):
    if event['RequestType'] == "Delete":
        send(event, context, "SUCCESS", {})
        return
    app_name = event['ResourceProperties']['AppName']
    app_names = map(lambda x: x['ApplicationName'], client.list_applications()['ApplicationSummaries'])
    if app_name not in app_names:
        send(event, context, "SUCCESS", {})
        return
    app_details = client.describe_application(ApplicationName=app_name)
    app_status = app_details['ApplicationDetail']['ApplicationStatus']
    input_id = app_details['ApplicationDetail']['InputDescriptions'][0]['InputId']
    if app_status == "READY":
        response = client.start_application(
            ApplicationName=app_name,
            InputConfigurations=[
                {
                    'Id': str(input_id),
                    'InputStartingPositionConfiguration': {
                        'InputStartingPosition': 'NOW'
                    }
                },
            ]
            )
        send(event, context, "SUCCESS", {})
    else:
        send(event, context, "FAILED", {})
        
def send(event, context, responseStatus, responseData, physicalResourceId=''):
    responseUrl = event['ResponseURL']
    responseBody = {}
    responseBody['Status'] = responseStatus
    responseBody['Reason'] = 'See the details in CloudWatch Log Stream: ' + context.log_stream_name
    responseBody['PhysicalResourceId'] = physicalResourceId or context.log_stream_name
    responseBody['StackId'] = event['StackId']
    responseBody['RequestId'] = event['RequestId']
    responseBody['LogicalResourceId'] = event['LogicalResourceId']
    responseBody['Data'] = responseData
    json_responseBody = json.dumps(responseBody)
    headers = {
        'content-type' : '', 
        'content-length' : str(len(json_responseBody))
    }
    try:
        response = requests.put(responseUrl,
                                data=json_responseBody,
                                headers=headers)
        print "Status code: " + response.reason
    except Exception as e:
        print "send(..) failed executing requests.put(..): " + str(e)

File: ka_start/ka-start.js
'use strict';
var AWS = require('aws-sdk');

var kinesisanalytics = new AWS.KinesisAnalytics();

exports.handler = (event, context, callback) => {
	//var app_details = client.describe_application(ApplicationName=app_name)
    if (event.RequestType == "Create") {
    	var params = {
    		ApplicationName: event.ResourceProperties.AppName
    	};
    	kinesisanalytics.describeApplication(params, (err,data) => {
    		if(err){
    			console.log('Error calling describeApplication: ' + err);
    			sendResponse(event, context, "FAILED"); 
    			callback(err, "Unable to describe application: " + event.ResourceProperties.AppName );
    		} else {
    			var input_id = data.ApplicationDetail.InputDescriptions[0].InputId;
                var app_status =  data.ApplicationDetail.ApplicationStatus;
                if(app_status=="READY") {
                    var start_params = {
                        ApplicationName: event.ResourceProperties.AppName,
                        InputConfigurations : [ {
                            Id: input_id,
                            InputStartingPositionConfiguration : { InputStartingPosition: 'NOW' }
                        } ]
                    };
                    kinesisanalytics.startApplication(start_params, (err,data) => {
                        if(err) { 
                            console.log('Error starting Kinesis Application: ' + err);
                            sendResponse(event,context,"FAILED");
                            callback(err,"FAILED");
                        }
                        else {
                            sendResponse(event,context,"SUCCESS");
                            callback(null,"SUCCESS");
                        }
                    });
                } else {
                    console.log('Error: Kinesis Application not READY was ' + app_status );
                    callback("Not Ready","FAILED");
                }
    		}
    	});
    } else { //action only needed on Create
        sendResponse(event,context,"SUCCESS");
        callback(null,"SUCCESS"); 
    }        
};

function sendResponse(event, context, responseStatus, responseData) {

    var responseBody = JSON.stringify({
        Status: responseStatus,
        Reason: "See the details in CloudWatch Log Stream: " + context.logStreamName,
        PhysicalResourceId: context.logStreamName,
        StackId: event.StackId,
        RequestId: event.RequestId,
        LogicalResourceId: event.LogicalResourceId,
        Data: responseData
    });


    console.log("RESPONSE BODY:\n", responseBody);

    var https = require("https");
    var url = require("url");

    var parsedUrl = url.parse(event.ResponseURL);
    var options = {
        hostname: parsedUrl.hostname,
        port: 443,
        path: parsedUrl.path,
        method: "PUT",
        headers: {
            "content-type": "",
            "content-length": responseBody.length
        }
    };

    console.log("SENDING RESPONSE...\n");

    var request = https.request(options, function(response) {
        console.log("STATUS: " + response.statusCode);
        console.log("HEADERS: " + JSON.stringify(response.headers));
    });

    request.on("error", function(error) {
        console.log("sendResponse Error:" + error);
    });

    // write data to request body
    request.write(responseBody);
    request.end();
};

